docs/advanced-features/react-18/server-components.md

```diff
@@ -0,0 +1,124 @@
+# React Server Components (Alpha)
+
+Server Components allow us to render React components on the server. This is fundamentally different from server-side rendering where you're pre-generating HTML on the server. With Server Components, there's **zero client-side JavaScript needed,** making page rendering faster. This improves the user experience of your application, pairing the best parts of server-rendering with client-side interactivity.
+
+### Enable React Server Components
+
+To use React Server Components, ensure you have React 18 installed. Then, turn on the `concurrentFeatures` and `serverComponents` options in `next.config.js`:
+
+```jsx
+// next.config.js
+module.exports = {
+  experimental: {
+    concurrentFeatures: true,
+    serverComponents: true,
+  },
+}
+```
+
+Next, if you already have customized `pages/_document` component, you need to remove the `getInitialProps` static method and the `getServerSideProps` export if there’s any, otherwise it won't work with server components. If no custom Document component is provided, Next.js will fallback to a basic functional document component. Check [next/document support](#next/document) for more details.
+
+Then, you can start using React Server Components. [See our example](https://github.com/vercel/next-rsc-demo) for more information.
+
+### Server Components Convention
+
+To run a component on the server, append `.server.js` to the end of the filename. For example `./pages/home.server.js` is a Server Component.
```

Comment:
```suggestion
To run a component on the server, append `.server.js` to the end of the filename. For example `./components/home.server.js` is a Server Component.
```

(not sure about this)

There’s `./components/avatar.client.js` in the next paragraph.

Reply:
Is it worth mentioning jsx or tsx?